flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -642,6 +654,35 @@ public static String getDynamicProperties(Configuration baseConfig, Configuratio
 		return String.join(" ", newAddedConfigs);
 	}
 
+	/**
+	 * Escape all the dynamic property values.
+	 * For unix-like OS(Linux, MacOS, FREE_BSD, etc.), each value will be surrounded with single quotes. This works for
+	 * all chars except single quote itself. To escape the single quote, close the quoting before it, insert the escaped
+	 * single quote, and then re-open the quoting. For example, the value is foo'bar and the escaped value is
+	 * 'foo'\''bar'.
```

Comment:
Would be good to add the stackoverflow reference for how we escape strings for Unix like systems.

Reply:
https://stackoverflow.com/questions/15783701/which-characters-need-to-be-escaped-when-using-bash

I will add the above url to the document.